DMS plugin crashed

<!--
Hi there,

Thank you for opening an issue. Please note that we try to keep the issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

If your issue relates to Terraform, please open it in the terraform's repository at https://github.com/hashicorp/terraform.
-->

### Terraform Version
<!---
Run `terraform -v` to show the version, and paste the result between the ``` marks below.
Terraform v1.3.2
If you are not running the latest version of Terraform, please try upgrading because your issue may have already been fixed.
-->

```
...
```

### Terraform Configuration Files
<!--
Paste the relevant parts of your Terraform configuration between the ``` marks below.

For large Terraform configs, please use a service like Dropbox and share a link to the ZIP file. For security, you can also encrypt the files using our GPG public key.
-->

resource "huaweicloud_networking_secgroup" "secgroup" {
  name        = var.security_group_name
  description = "terraform security group rabbit"
}

data "huaweicloud_dms_product" "product_1" {
  engine        = "rabbitmq"
  instance_type = "single"
  version       = "3.7.17"
  node_num      = 1
  
}

resource "huaweicloud_dms_rabbitmq_instance" "test" {
  name              = "test-api"
  product_id        = data.huaweicloud_dms_product.product_1.id
  engine_version    = data.huaweicloud_dms_product.product_1.version
  storage_spec_code = data.huaweicloud_dms_product.product_1.storage_spec_code

  vpc_id             = "111111"
  network_id         = "22222"
  security_group_id  = huaweicloud_networking_secgroup.secgroup.id
  availability_zones = ["zone"]

  access_user = "user"
  password    = "pass"

```hcl
...
```

### Debug Output
<!--
Full debug output can be obtained by running Terraform with the environment variable `TF_LOG=trace`. Please create a GitHub Gist containing the debug output. Please do _not_ paste the debug output in the issue, since debug output is long.


huaweicloud_dms_rabbitmq_instance.cadastro: Creating...
╷
│ Error: Plugin did not respond
│ 
│   with huaweicloud_dms_rabbitmq_instance.cadastro,
│   on cadastro.tf line 16, in resource "huaweicloud_dms_rabbitmq_instance" "test":
│   16: resource "huaweicloud_dms_rabbitmq_instance" "test" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.

Debug output may contain sensitive information. Please review it before posting publicly, and if you are concerned feel free to encrypt the files using the HashiCorp security public key.
-->

### Crash Output
<!--
If the console output indicates that Terraform crashed, please share a link to a GitHub Gist containing the output of the `crash.log` file.
-->

### Expected Behavior
<!--
What should have happened?
-->

### Actual Behavior
<!--
What actually happened?
-->

### Steps to Reproduce
<!--
Please list the full steps required to reproduce the issue, for example:
1. `terraform init`
2. `terraform apply`
-->

### Additional Context
when I try to create the object , It returned the msg error
Stack trace from the terraform-provider-huaweicloud_v1.41.1 plugin:

panic: runtime error: index out of range [0] with length 0

goroutine 31 [running]:
github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/dms.resourceDmsRabbitmqInstanceCreate({0x1de4748, 0xc0002d2c60}, 0x180eb00?, {0x1af9940?, 0xc000d33400})
    github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/dms/resource_huaweicloud_dms_rabbitmq_instance.go:259 +0x12ef
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc0005f6ee0, {0x1de46d8, 0xc00024d4c0}, 0x2?, {0x1af9940, 0xc000d33400})
    github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.0/helper/schema/resource.go:341 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0005f6ee0, {0x1de46d8, 0xc00024d4c0}, 0xc000df4820, 0xc0000ce400, {0x1af9940, 0xc000d33400})
    github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.0/helper/schema/resource.go:467 +0x705
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000010288, {0x1de46d8, 0xc00024d4c0}, 0xc0010265f0)
    github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.0/helper/schema/grpc_provider.go:977 +0xde5
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc00012ae00, {0x1de4780?, 0xc00101c360?}, 0xc00100c000)
    github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:603 +0x2ad
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x1a70b60?, 0xc00012ae00}, {0x1de4780, 0xc00101c360}, 0xc0002d2060, 0x0)
    github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00020ddc0, {0x1de87b8, 0xc00082ec00}, 0xc00026e100, 0xc000acdfb0, 0x2ae4fc0, 0x0)
    google.golang.org/grpc@v1.32.0/server.go:1194 +0xcab
google.golang.org/grpc.(*Server).handleStream(0xc00020ddc0, {0x1de87b8, 0xc00082ec00}, 0xc00026e100, 0x0)
    google.golang.org/grpc@v1.32.0/server.go:1517 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
    google.golang.org/grpc@v1.32.0/server.go:859 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
    google.golang.org/grpc@v1.32.0/server.go:857 +0x28a

Error: The terraform-provider-huaweicloud_v1.41.1 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

### References
<!--
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:

- #6017

-->

--- a/huaweicloud/services/acceptance/dms/data_source_huaweicloud_dms_kafka_instances_test.go
+++ b/huaweicloud/services/acceptance/dms/data_source_huaweicloud_dms_kafka_instances_test.go
@@ -17,7 +17,7 @@ func TestAccKafkaInstancesDataSource_basic(t *testing.T) {
 	dc3 := acceptance.InitDataSourceCheck("data.huaweicloud_dms_kafka_instances.query_3")
 	dc4 := acceptance.InitDataSourceCheck("data.huaweicloud_dms_kafka_instances.query_4")
 
-	resource.ParallelTest(t, resource.TestCase{
+	resource.Test(t, resource.TestCase{
 		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
 		ProviderFactories: acceptance.TestAccProviderFactories,
 		Steps: []resource.TestStep{
@@ -47,6 +47,7 @@ func TestAccKafkaInstancesDataSource_basic(t *testing.T) {
 
 func testAccKafkaInstancesDataSource_basic() string {
 	rName := acceptance.RandomAccResourceNameWithDash()
+	fuzzyName := rName[0 : len(rName)-1]
 
 	return fmt.Sprintf(`
 %s
@@ -56,11 +57,15 @@ data "huaweicloud_dms_kafka_instances" "query_0" {
     huaweicloud_dms_kafka_instance.test,
   ]
 
-  name        = "tf-acc-test"
+  name        = "%s"
   fuzzy_match = true
 }
 
 data "huaweicloud_dms_kafka_instances" "query_1" {
+  depends_on = [
+    huaweicloud_dms_kafka_instance.test,
+  ]
+  
   name = huaweicloud_dms_kafka_instance.test.name
 }
 
@@ -69,6 +74,10 @@ data "huaweicloud_dms_kafka_instances" "query_2" {
 }
 
 data "huaweicloud_dms_kafka_instances" "query_3" {
+  depends_on = [
+    huaweicloud_dms_kafka_instance.test,
+  ]
+  
   enterprise_project_id = huaweicloud_dms_kafka_instance.test.enterprise_project_id
 }
 
@@ -79,5 +88,5 @@ data "huaweicloud_dms_kafka_instances" "query_4" {
 
   status = "RUNNING"
 }
-`, testAccKafkaInstance_basic(rName))
+`, testAccKafkaInstance_basic(rName), fuzzyName)
 }

--- a/huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_kafka_instance_test.go
+++ b/huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_kafka_instance_test.go
@@ -33,7 +33,8 @@ func TestAccKafkaInstance_basic(t *testing.T) {
 		getKafkaInstanceFunc,
 	)
 
-	resource.ParallelTest(t, resource.TestCase{
+	// DMS instances use the tenant-level shared lock, the instances cannot be created or modified in parallel.
+	resource.Test(t, resource.TestCase{
 		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
 		ProviderFactories: acceptance.TestAccProviderFactories,
 		CheckDestroy:      rc.CheckResourceDestroy(),
@@ -84,7 +85,7 @@ func TestAccKafkaInstance_withEpsId(t *testing.T) {
 		getKafkaInstanceFunc,
 	)
 
-	resource.ParallelTest(t, resource.TestCase{
+	resource.Test(t, resource.TestCase{
 		PreCheck:          func() { acceptance.TestAccPreCheckEpsID(t) },
 		ProviderFactories: acceptance.TestAccProviderFactories,
 		CheckDestroy:      rc.CheckResourceDestroy(),
@@ -114,7 +115,7 @@ func TestAccKafkaInstance_compatible(t *testing.T) {
 		getKafkaInstanceFunc,
 	)
 
-	resource.ParallelTest(t, resource.TestCase{
+	resource.Test(t, resource.TestCase{
 		PreCheck:          func() { acceptance.TestAccPreCheckEpsID(t) },
 		ProviderFactories: acceptance.TestAccProviderFactories,
 		CheckDestroy:      rc.CheckResourceDestroy(),
@@ -144,7 +145,7 @@ func TestAccKafkaInstance_newFormat(t *testing.T) {
 		getKafkaInstanceFunc,
 	)
 
-	resource.ParallelTest(t, resource.TestCase{
+	resource.Test(t, resource.TestCase{
 		PreCheck:          func() { acceptance.TestAccPreCheckEpsID(t) },
 		ProviderFactories: acceptance.TestAccProviderFactories,
 		CheckDestroy:      rc.CheckResourceDestroy(),

--- a/huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_rabbitmq_instance_test.go
+++ b/huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_rabbitmq_instance_test.go
@@ -33,7 +33,8 @@ func TestAccDmsRabbitmqInstances_basic(t *testing.T) {
 		getDmsRabitMqInstanceFunc,
 	)
 
-	resource.ParallelTest(t, resource.TestCase{
+	// DMS instances use the tenant-level shared lock, the instances cannot be created or modified in parallel.
+	resource.Test(t, resource.TestCase{
 		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
 		ProviderFactories: acceptance.TestAccProviderFactories,
 		CheckDestroy:      rc.CheckResourceDestroy(),
@@ -49,14 +50,6 @@ func TestAccDmsRabbitmqInstances_basic(t *testing.T) {
 				),
 			},
 			{
-				ResourceName:      resourceName,
-				ImportState:       true,
-				ImportStateVerify: true,
-				ImportStateVerifyIgnore: []string{
-					"password",
-				},
-			},
-			{
 				Config: testAccDmsRabbitmqInstance_update(rName, updateName),
 				Check: resource.ComposeTestCheckFunc(
 					rc.CheckResourceExists(),
@@ -66,6 +59,14 @@ func TestAccDmsRabbitmqInstances_basic(t *testing.T) {
 					resource.TestCheckResourceAttr(resourceName, "tags.key1", "value"),
 					resource.TestCheckResourceAttr(resourceName, "tags.owner", "terraform_update"),
 				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateVerifyIgnore: []string{
+					"password", "used_storage_space",
+				},
 			},
 		},
 	})
@@ -81,7 +82,7 @@ func TestAccDmsRabbitmqInstances_withEpsId(t *testing.T) {
 		getDmsRabitMqInstanceFunc,
 	)
 
-	resource.ParallelTest(t, resource.TestCase{
+	resource.Test(t, resource.TestCase{
 		PreCheck:          func() { acceptance.TestAccPreCheckEpsID(t) },
 		ProviderFactories: acceptance.TestAccProviderFactories,
 		CheckDestroy:      rc.CheckResourceDestroy(),
@@ -111,7 +112,7 @@ func TestAccDmsRabbitmqInstances_compatible(t *testing.T) {
 		getDmsRabitMqInstanceFunc,
 	)
 
-	resource.ParallelTest(t, resource.TestCase{
+	resource.Test(t, resource.TestCase{
 		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
 		ProviderFactories: acceptance.TestAccProviderFactories,
 		CheckDestroy:      rc.CheckResourceDestroy(),
@@ -131,7 +132,7 @@ func TestAccDmsRabbitmqInstances_compatible(t *testing.T) {
 				ImportState:       true,
 				ImportStateVerify: true,
 				ImportStateVerifyIgnore: []string{
-					"password",
+					"password", "used_storage_space",
 				},
 			},
 		},
@@ -148,7 +149,7 @@ func TestAccDmsRabbitmqInstances_single(t *testing.T) {
 		getDmsRabitMqInstanceFunc,
 	)
 
-	resource.ParallelTest(t, resource.TestCase{
+	resource.Test(t, resource.TestCase{
 		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
 		ProviderFactories: acceptance.TestAccProviderFactories,
 		CheckDestroy:      rc.CheckResourceDestroy(),
@@ -165,17 +166,21 @@ func TestAccDmsRabbitmqInstances_single(t *testing.T) {
 
 func testAccDmsRabbitmqInstance_Base(rName string) string {
 	return fmt.Sprintf(`
-
-data "huaweicloud_vpc" "test" {
-  name = "vpc-default"
+resource "huaweicloud_vpc" "test" {
+  name        = "%[1]s"
+  cidr        = "192.168.0.0/24"
+  description = "Test for DMS RabbitMQ"
 }
 
-data "huaweicloud_vpc_subnet" "test" {
-  name = "subnet-default"
+resource "huaweicloud_vpc_subnet" "test" {
+  name       = "%[1]s"
+  cidr       = "192.168.0.0/24"
+  gateway_ip = "192.168.0.1"
+  vpc_id     = huaweicloud_vpc.test.id
 }
 
 resource "huaweicloud_networking_secgroup" "test" {
-  name        = "%s"
+  name        = "%[1]s"
   description = "secgroup for rabbitmq"
 }
 
@@ -205,8 +210,8 @@ resource "huaweicloud_dms_rabbitmq_instance" "test" {
   name        = "%s"
   description = "rabbitmq test"
   
-  vpc_id             = data.huaweicloud_vpc.test.id
-  network_id         = data.huaweicloud_vpc_subnet.test.id
+  vpc_id             = huaweicloud_vpc.test.id
+  network_id         = huaweicloud_vpc_subnet.test.id
   security_group_id  = huaweicloud_networking_secgroup.test.id
   availability_zones = [
     data.huaweicloud_availability_zones.test.names[0]
@@ -235,8 +240,8 @@ resource "huaweicloud_dms_rabbitmq_instance" "test" {
   name        = "%s"
   description = "rabbitmq test update"
 
-  vpc_id             = data.huaweicloud_vpc.test.id
-  network_id         = data.huaweicloud_vpc_subnet.test.id
+  vpc_id             = huaweicloud_vpc.test.id
+  network_id         = huaweicloud_vpc_subnet.test.id
   security_group_id  = huaweicloud_networking_secgroup.test.id
   availability_zones = [
     data.huaweicloud_availability_zones.test.names[0]
@@ -266,8 +271,8 @@ resource "huaweicloud_dms_rabbitmq_instance" "test" {
   description           = "rabbitmq test"
   enterprise_project_id = "%s"
 
-  vpc_id             = data.huaweicloud_vpc.test.id
-  network_id         = data.huaweicloud_vpc_subnet.test.id
+  vpc_id             = huaweicloud_vpc.test.id
+  network_id         = huaweicloud_vpc_subnet.test.id
   security_group_id  = huaweicloud_networking_secgroup.test.id
   availability_zones = [
     data.huaweicloud_availability_zones.test.names[0],
@@ -299,8 +304,8 @@ resource "huaweicloud_dms_rabbitmq_instance" "test" {
   name        = "%s"
   description = "rabbitmq test"
   
-  vpc_id            = data.huaweicloud_vpc.test.id
-  network_id        = data.huaweicloud_vpc_subnet.test.id
+  vpc_id            = huaweicloud_vpc.test.id
+  network_id        = huaweicloud_vpc_subnet.test.id
   security_group_id = huaweicloud_networking_secgroup.test.id
   available_zones   = [data.huaweicloud_dms_az.test.id]
 
@@ -324,35 +329,13 @@ resource "huaweicloud_dms_rabbitmq_instance" "test" {
 func testAccDmsRabbitmqInstance_single(rName string) string {
 	randPwd := fmt.Sprintf("%s!#%d", acctest.RandString(5), acctest.RandIntRange(0, 999))
 	return fmt.Sprintf(`
-data "huaweicloud_availability_zones" "test" {}
+%[1]s
 
-data "huaweicloud_dms_product" "test" {
+data "huaweicloud_dms_product" "single" {
   engine           = "rabbitmq"
   instance_type    = "single"
   version          = "3.7.17"
   node_num         = 1
-}
-
-resource "huaweicloud_vpc" "test" {
-  name        = "%[1]s"
-  cidr        = "192.168.0.0/16"
-  description = "Test for DMS RabbitMQ"
-}
-
-locals {
-  subnet_cidr = cidrsubnet(huaweicloud_vpc.test.cidr, 4, 1)
-}
-
-resource "huaweicloud_vpc_subnet" "test" {
-  name       = "%[1]s"
-  cidr       = local.subnet_cidr
-  gateway_ip = cidrhost(local.subnet_cidr, 1)
-  vpc_id     = huaweicloud_vpc.test.id
-}
-
-resource "huaweicloud_networking_secgroup" "test" {
-  name        = "%[1]s"
-  description = "The security group for DMS Kafka"
 }
 
 resource "huaweicloud_dms_rabbitmq_instance" "test" {
@@ -360,18 +343,18 @@ resource "huaweicloud_dms_rabbitmq_instance" "test" {
     data.huaweicloud_availability_zones.test.names[0],
   ]
 
-  name              = "%[1]s"
+  name              = "%[2]s"
   vpc_id            = huaweicloud_vpc.test.id
   network_id        = huaweicloud_vpc_subnet.test.id
   security_group_id = huaweicloud_networking_secgroup.test.id
 
-  product_id        = data.huaweicloud_dms_product.test.id
-  engine_version    = data.huaweicloud_dms_product.test.version
-  storage_spec_code = data.huaweicloud_dms_product.test.storage_spec_code
-  storage_space     = data.huaweicloud_dms_product.test.storage
+  product_id        = data.huaweicloud_dms_product.single.id
+  engine_version    = data.huaweicloud_dms_product.single.version
+  storage_spec_code = data.huaweicloud_dms_product.single.storage_spec_code
+  storage_space     = data.huaweicloud_dms_product.single.storage
 
   access_user = "root"
-  password    = "%[2]s"
+  password    = "%[3]s"
 }
-`, rName, randPwd)
+`, testAccDmsRabbitmqInstance_Base(rName), rName, randPwd)
 }

--- a/huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go
+++ b/huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go
@@ -852,22 +852,20 @@ func resizeInstance(ctx context.Context, d *schema.ResourceData, meta interface{
 
 	opts := instances.ResizeInstanceOpts{}
 	if _, ok := d.GetOk("product_id"); ok {
-		var product *products.Detail
 		if engineType == "kafka" {
-			product, err = getKafkaProductDetail(config, d)
+			product, err := getKafkaProductDetail(config, d)
+			if err != nil {
+				return fmtp.Errorf("change storage_space failed, error querying product detail: %s", err)
+			}
+			opts.NewSpecCode = product.SpecCode
 		} else {
-			product, err = getRabbitMQProductDetail(config, d)
+			product, err := getRabbitMQProductDetail(config, d)
+			if err != nil {
+				return fmtp.Errorf("change storage_space failed, error querying product detail: %s", err)
+			}
+			opts.NewSpecCode = product.SpecCode
 		}
-		if err != nil || product == nil {
-			return fmtp.Errorf("change storage_space failed, error querying product detail: %s", err)
-		}
-		logp.Printf("[DEBUG] Get DMS product detail: %#v", product)
-
 		opts.NewStorageSpace = d.Get("storage_space").(int)
-		opts.NewSpecCode = product.SpecCode
-		if engineType == "rabbitmq" {
-			opts.NewSpecCode = product.ProductInfos[0].SpecCode
-		}
 	} else {
 		opts.NewSpecCode = d.Get("storage_spec_code").(string)
 		opts.NewStorageSpace = d.Get("storage_space").(int)
@@ -917,7 +915,7 @@ func resourceDmsKafkaInstanceDelete(ctx context.Context, d *schema.ResourceData,
 	logp.Printf("[DEBUG] Waiting for instance (%s) to delete", d.Id())
 
 	stateConf := &resource.StateChangeConf{
-		Pending:      []string{"DELETING", "RUNNING"},
+		Pending:      []string{"DELETING", "RUNNING", "ERROR"}, // Status may change to ERROR on deletion.
 		Target:       []string{"DELETED"},
 		Refresh:      DmsKafkaInstanceStateRefreshFunc(dmsV2Client, d.Id()),
 		Timeout:      d.Timeout(schema.TimeoutDelete),

--- a/huaweicloud/services/dms/resource_huaweicloud_dms_rabbitmq_instance.go
+++ b/huaweicloud/services/dms/resource_huaweicloud_dms_rabbitmq_instance.go
@@ -193,7 +193,7 @@ func ResourceDmsRabbitmqInstance() *schema.Resource {
 	}
 }
 
-func getRabbitMQProductDetail(config *config.Config, d *schema.ResourceData) (*products.Detail, error) {
+func getRabbitMQProductDetail(config *config.Config, d *schema.ResourceData) (*products.ProductInfo, error) {
 	productRsp, err := getProducts(config, config.GetRegion(d), "rabbitmq")
 	if err != nil {
 		return nil, fmtp.Errorf("error querying product detail, please check product_id, error: %s", err)
@@ -207,22 +207,27 @@ func getRabbitMQProductDetail(config *config.Config, d *schema.ResourceData) (*p
 			continue
 		}
 		for _, v := range ps.Values {
-			for _, p := range v.Details {
+			for _, detail := range v.Details {
 				// All informations of product for single instance type and the kafka engine type are stored in the
 				// detail structure.
 				if v.Name == "single" {
-					if p.ProductID == productID {
-						return &p, nil
+					if detail.ProductID == productID {
+						return &products.ProductInfo{
+							Storage:          detail.Storage,
+							ProductID:        detail.ProductID,
+							SpecCode:         detail.SpecCode,
+							IOs:              detail.IOs,
+							AvailableZones:   detail.AvailableZones,
+							UnavailableZones: detail.UnavailableZones,
+						}, nil
 					}
 				} else {
-					for _, pi := range p.ProductInfos {
-						if pi.ProductID == productID {
-							p.ProductInfos = []products.ProductInfo{pi}
-							return &p, nil
+					for _, product := range detail.ProductInfos {
+						if product.ProductID == productID {
+							return &product, nil
 						}
 					}
 				}
-
 			}
 		}
 	}
@@ -256,10 +261,9 @@ func resourceDmsRabbitmqInstanceCreate(ctx context.Context, d *schema.ResourceDa
 		if err != nil || product == nil {
 			return fmtp.DiagErrorf("query DMS RabbimtMQ product failed: %s", err)
 		}
-		space := product.ProductInfos[0].Storage
-		defaultStorageSpace, err := strconv.ParseInt(space, 10, 32)
+		defaultStorageSpace, err := strconv.ParseInt(product.Storage, 10, 32)
 		if err != nil {
-			return fmtp.DiagErrorf("Parse storage capacity to int error, %v: %s", space, err)
+			return fmtp.DiagErrorf("Parse storage capacity to int error, %v: %s", product.Storage, err)
 		}
 		storageSpace = int(defaultStorageSpace)
 	}

